common/pango.c

```diff
@@ -84,12 +87,12 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
 
 void get_text_size(cairo_t *cairo, const char *font, int *width, int *height,
 		int *baseline, double scale, bool markup, const char *fmt, ...) {
-	static char buf[2048];
+	char buf[max_chars + sizeof(overflow)];
 
 	va_list args;
 	va_start(args, fmt);
-	if (vsnprintf(buf, 2048, fmt, args) >= 2048) {
-		strcpy(buf, "[buffer overflow]");
+	if (vsnprintf(buf, max_chars, fmt, args) >= max_chars) {
+		strcat(buf, overflow);
```

Comment:
Sorry to pick nits here, but can you make the size of the buffer `max_chars`, then strcpy the overflow text to the end? Something like this:

```c
char buf[max_chars];
// ...
strcpy(&buf[sizeof(buf) - sizeof(overflow)], overflow);
```

Use sizeof(buf) as well for vsnprintf.